R/path_linters.R

```diff
@@ -101,41 +100,37 @@ is_valid_long_path <- function(path, lax = FALSE) {
 }
 
 
-split_path <- function(path, sep = "/|\\\\") {
+split_paths <- function(path, sep = "/|\\\\") {
   if (!is.character(path)) {
     stop("argument 'path' should be a character vector")
   }
   if (!is.character(sep) || length(sep) != 1L || !nzchar(sep)) {
     stop("argument 'sep' should be a non-empty regular expression character string")
   }
-  Map(
-    function(dirs, prefix) {
-      # add root dir if needed
-      i <- 1L
-      for (dir in seq_along(dirs)) {
-        if (!nzchar(dirs[[i]])) {
-          i <- i + 1L
-        } else {
-          break
-        }
-      }
-      i <- i - 1L
-      if (i > 0L) {
-        dirs <- c(paste0(rep(prefix, i), collapse = ""), tail(dirs, -i))
-      }
-      # add // to protocols (like http, smb, ...)
-      if (length(dirs)) {
-        l <- nchar(dirs[[1L]])
-        if (l > 2L && substr(dirs[[1L]], l, l) == ":") {
-          dirs[[1L]] <- paste0(dirs[[1L]], "//")
-        }
-      }
-      # remove empty dirs
-      dirs[nzchar(dirs)]
-    },
-    strsplit(path, sep),
-    substr(path, 1L, 1L)
-  )
+  Map(split_path, strsplit(path, sep), substr(path, 1L, 1L))
+}
+
+split_path <- function(dirs, prefix) {
+  # add root dir if needed
+  i <- 1L
+  for (dir in seq_along(dirs)) {
+    if (nzchar(dirs[[i]])) {
+      break
+    }
+    i <- i + 1L
+  }
+  i <- i - 1L
```

Comment:
So `i` is the last index such that `!nzchar(dirs[[i]])`.
But `nzchar` is vectorized, so

```r
i <- match(TRUE, nzchar(dirs)) - 1L
```

right?

Reply:
basically -- need to account for `dirs = ""` too. done.